src/types.rs

```diff
@@ -627,24 +629,32 @@ enum AbsurdComparisonResult {
     InequalityImpossible,
 }
 
+enum Rel {
+    Lt,
+    Le,
+}
+
+// Put the expression in the form lhs < rhs or lhs <= rhs.
+fn normalize_comparison<'a>(op: BinOp_, lhs: &'a Expr, rhs: &'a Expr)
+                                -> Option<(Rel, &'a Expr, &'a Expr)> {
+    match op {
+        BiLt => Some((Rel::Lt, lhs, rhs)),
+        BiLe => Some((Rel::Le, lhs, rhs)),
+        BiGt => Some((Rel::Lt, rhs, lhs)),
+        BiGe => Some((Rel::Le, rhs, lhs)),
+        _ => None,
+    }
+}
```

Comment:
I didn't actually intend for it to be a doc comment, hence the lack of markdown. I'll move it over and make that change.